lib/Interface/core/Container.cpp

```diff
@@ -211,7 +211,7 @@ template <typename T, std::size_t N> void MemRef<T, N>::setStrides() {
   strides[N - 1] = 1;
   if (N < 2)
     return;
-  for (int i = N - 2; i >= 0; i--) {
+  for (int i = static_cast<int>(N - 2); i >= 0; i--) {
```

Comment:
What warning will be reported now?
I think we can change `int i = ...` to `size_t i = ...`.

Reply:
if using origin code 'int i = N - 2 ', the error is "warning: overflow in conversion from ‘long unsigned int’ to ‘int’ changes value from ‘18446744073709551615’ to ‘-1’ [-Woverflow]".
![image](https://user-images.githubusercontent.com/96034746/208237193-a28ca79c-d561-451c-a068-47fa242cfbc0.png)
if replace 'int i = N - 2' with ''size_t i = N - 2'',it can build successfully. But, the executable file "vectorization-benchmark" will cause 'segmentation fault'.
![image](https://user-images.githubusercontent.com/96034746/208237494-0274ba57-ae43-4ac7-949d-75cd69009bf7.png)

Reply:
This seems weird, I'll try to reproduce the segmentation fault error.

Reply:
@Panhaolin2001 , can you share your local `buddy-mlir` and `buddy-benchmark`'s latest git commit id, which results in the above mentioned `segmentation fault`.

Reply:
@meshtag The case that caused this error is a PR in buddy- benchmark ：_https://github.com/buddy-compiler/buddy-benchmark/pull/50_
BTW, the MemRef I used whose dimension is one.

Reply:
The error occurred in two other benchmarks in the Vectoration folder. Both had a MemRef dimension of 2. When this code was executed:
```
for (std::size_t i = N - 2; i >= 0; i--) {
    strides[i] = strides[i + 1] * sizes[i + 1];
  }
```
The initial value of i was 0, and after a round of loops i became -1. After converting i to an unsigned integer (i = 18446744073709551615), the loop condition was met and the loop body was executed again.
I think there are two solutions to this issue. The first is to cast the size_t type of N-1 to int. The second is to change the loop index:
```
for (std::size_t i = N - 1; i > 0; i--) {
    strides[i - 1] = strides[i] * sizes[i];
  }

```

Reply:
`std::size_t` actually corresponds to unsigned integer type. 
(Ref : https://en.cppreference.com/w/cpp/types/size_t#:~:text=std%3A%3Asize_t%20is%20the,(since%20C%2B%2B11).)
The last iteration of this loop `for (std::size_t i = N - 2; i >= 0; --i)` might actually lead to the assignment of a negative value (-1) to the variable `i`. This will definitely cause issues. 

@Panhaolin2001, can you check if this is the case by making some changes to verify this with your PR on your local system.

Reply:
Changing the loop index (with a comment on that location explaining the reason) makes more sense to me.

Reply:
OK,thanks, i will change the loop index now.